src/viv_render.c

```diff
@@ -221,9 +263,13 @@ static void viv_render_xdg_view(struct wlr_renderer *renderer, struct viv_view *
     clock_gettime(CLOCK_MONOTONIC, &now);
 
     // Note this renders both the toplevel and any popups
-    struct wlr_box actual_geometry = { 0 };
     struct wlr_box *target_geometry = &view->target_box;
-    wlr_xdg_surface_get_geometry(view->xdg_surface, &actual_geometry);
+
+    // Update floating view sizes, as the client has control over it
+    if (view->is_floating || (view->workspace->fullscreen_view == view)) {
+        viv_view_match_target_box_with_surface_geometry(view);
```

Comment:
Aha, this seems to be producing a target box that doesn't quite match the actual geometry

Reply:
I think the issue is that `match_target_box_with_surface_geometry` adds to the target_box the border width, but actually the target box should be the size of the surface and the borders are drawn around (outside) it. I'm pushing the fix to my branch, I'll test for a bit and merge if all seems good.

I'm not sure it's ideal to reset the target box all the time like this but I suppose the cost is pretty trivial so not an issue to remove later.